engine/consensus/approvals/assignment_collector_tree.go

```diff
@@ -261,11 +261,18 @@ func (t *AssignmentCollectorTree) GetOrCreateCollector(result *flow.ExecutionRes
 		}, nil
 	}
 
+	// leveled forest doesn't treat this case as error, we shouldn't create collectors
+	// for vertexes lower that forest.LowestLevel
+	if vertex.Level() < t.forest.LowestLevel {
+		return nil, fmt.Errorf("could not add collector with height lower than the lowest level")
```

Comment:
I am not sure, but some thoughts:
* I feel due to concurrent processing, it could happen that we prune the assignment collector while in another thread adding a collector for the pruned height. 
* I don't think this would be a fatal error, right (?)

### Suggestion:

* use sentinel `engine.OutdatedInputError` here: 
   ```suggestion
		return nil, engine.NewOutdatedInputErrorf("cannot add collector because its height %d is smaller than the lowest height %d", vertex.Level(), t.forest.LowestLevel)
   ```
   `sealing.Core` already checks for this error (see code [here](https://github.com/onflow/flow-go/blob/321088900138ad05e010cd78333f05bad7dd5368/engine/consensus/sealing/core.go#L301-L310))

* please also add corresponding test to [`core_test.go`](https://github.com/onflow/flow-go/blob/321088900138ad05e010cd78333f05bad7dd5368/engine/consensus/sealing/core_test.go) that checks that the expected sentinel error is returned